### PR TITLE
Made Eligibility API to fail gracefully

### LINKED ISSUE #2517: Eligibility API errors when invalid isbn
Relates to #2194

This is unrelated to these changes, but if eligibility API given bad isbn, fails ungracefully:
https://dev.openlibrary.org/sponsorship/eligibility/97802992042042?debug=true

### Evidence / Screenshot (if possible)

```
<type 'exceptions.AttributeError'> at /sponsorship/eligibility/97802992042042
'NoneType' object has no attribute 'get_isbn13'

/opt/openlibrary/deploys/openlibrary/openlibrary/openlibrary/core/sponsorships.py in qualifies_for_sponsorship

edition.isbn = edition.get_isbn13() 

/opt/openlibrary/deploys/openlibrary/openlibrary/openlibrary/plugins/openlibrary/api.py in GET
        return simplejson.dumps(qualifies_for_sponsorship(edition)) 
```
### Relevant url?

https://dev.openlibrary.org/sponsorship/eligibility/97802992042042?debug=true

### Expectation

Should check `if edition` and gracefully error accordingly.

### Details

- **Logged in (Y/N)?** 
- **Browser type/version?** 
- **Operating system?**

### PR DESCRIPTION
**Pull request by tabshaikh**
Closes #2517

### Technical

Checked if not an eligible edition should throw an error

### Testing

Go to https://dev.openlibrary.org/sponsorship/eligibility/97802992042042 and check if the API throws an error gracefully

### Stakeholders

@mekarpeles if you have the bandwidth could you please review this :)

### CODE REVIEW

File: openlibrary/plugins/openlibrary/api.py [hunk 1]
```diff
@@ -308,8 +308,11 @@ def GET(self, _id):
             web.ctx.site.get('/books/%s' % _id)
             if re.match(r'OL[0-9]+M', _id)
             else models.Edition.from_isbn(_id)
-            
         )
+        if not edition:
+            return simplejson.dumps({
+                'error': 'Invalid ISBN 13.'
```
Comment (hornc) [inline openlibrary/plugins/openlibrary/api.py]: What is the response code here? It looks like it will  be a 200 OK, it should be a HTTP 400 response.

Comment (tabshaikh) [reply openlibrary/plugins/openlibrary/api.py]: Yes the response should be a HTTP 400 response. Will ask @mekarpeles to test on dev and give feedback

Comment (tabshaikh) [reply openlibrary/plugins/openlibrary/api.py]: @hornc I have updated the pr to give 400 Bad Request error

**Comment by cdrini** [comment]
It's a _better_ error, but the JSON error object isn't being served; I'm getting the literal text "400 Bad Request" in the response from https://dev.openlibrary.org/sponsorship/eligibility/97802992042042

**Comment by mekarpeles** [comment]
@tabshaikh , please see @cdrini's last link https://dev.openlibrary.org/sponsorship/eligibility/97802992042042 -- it looks like there are still problematic cases. :)

**Comment by tabshaikh** [comment]
Thanks @mek resolving it :)

**Comment by tabshaikh** [comment]
Updated pr #3168

**Pull request closed without merging**